pandas/tests/indexes/test_interval_new.py

```diff
@@ -0,0 +1,310 @@
+    def test_get_indexer_with_interval(self):
+
+        index = IntervalIndex.from_tuples(
+            [(0, 2.5), (1, 3), (2, 4)], closed='right')
+
+        # single queries
+        queries = [Interval(1, 3, closed='right'),
+                   Interval(1, 3, closed='left'),
+                   Interval(1, 3, closed='both'),
+                   Interval(1, 3, closed='neither'),
+                   Interval(1, 4, closed='right'),
+                   Interval(0, 4, closed='right'),
+                   Interval(1, 2, closed='right')]
+        expected = [1, -1, -1, -1, -1, -1, -1]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer([query])
+            expect = np.array([expected_result], dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+        # multiple queries
+        queries = [
+            [Interval(2, 4, closed='right'), Interval(1, 3, closed='right')],
+            [Interval(1, 3, closed='right'), Interval(0, 2, closed='right')],
+            [Interval(1, 3, closed='right'), Interval(1, 3, closed='left')],
+            index]
+        expected = [[2, 1], [1, -1], [1, -1], [0, 1, 2]]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer(query)
+            expect = np.array(expected_result, dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_with_ints_and_floats(self):
+
+        index = IntervalIndex.from_tuples(
+            [(0, 1), (1, 2), (3, 4)], closed='right')
+
+        # single queries
+        queries = [-0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5]
+        expected = [-1, -1, 0, 0, 1, 1, -1, -1, 2, 2, -1]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer([query])
+            expect = np.array([expected_result], dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+        # multiple queries
+        queries = [[1, 2], [1, 2, 3], [1, 2, 3, 4], [1, 2, 3, 4, 2]]
+        expected = [[0, 1], [0, 1, -1], [0, 1, -1, 2], [0, 1, -1, 2, 1]]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer(query)
+            expect = np.array(expected_result, dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+        # TODO: @shoyer believes this should raise, master branch doesn't
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_non_unique_with_ints_and_floats(self):
+
+        index = IntervalIndex.from_tuples(
+            [(0, 2.5), (1, 3), (2, 4)], closed='left')
+
+        # single queries
+        queries = [-0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5]
+        expected = [(Int64Index([], dtype='int64'), np.array([0]))
```

Comment:
same

anytime you have a list of queries / expected, need to parametrize this, it is **way simpler** to debug this